SpatialGDK/Source/SpatialGDK/Public/Interop/ActorSystem.h

```diff
@@ -38,14 +38,22 @@ struct ActorData
 
 struct FObjectRepNotifies
 {
-	FObjectRepNotifies(const TWeakObjectPtr<UObject> WeakObjectPtr)
-		: Object(WeakObjectPtr)
-	{
-	}
-	TWeakObjectPtr<UObject> Object;
+	UObject* Object;
 	TArray<GDK_PROPERTY(Property)*> RepNotifies;
```

Comment:
We could change this to a TSet, there is theoretical worse performance in ComponentReader::ApplySchemaObject which uses AddUnique to add items to the TArray. This is because the TArray will now be storing FProperties for all properties on an object, instead of only those on a single spatial component.

(however this would probably need to be later converted to a list so it can be sorted, I think set lower performance at low number of items + this means TArray is still better)

Reply:
I would guess TArray would be faster in all situations, unless its very very large. Also TSet can be inefficient when storing small things.

Reply:
yep, I guess we'll find if this is an issue if NFRs take a hit (hopefully they're real world enough for this!)